### DISCUSSION TITLE
Why INVOCATION.Request is sequential?

### DISCUSSION THREAD

**Discussion by SunFulong** [wamp-proto/wamp-proto#532] [General]
At current spec, the INVOCATION.Request is the only one sequential ID which issued by router(dealer), and only one need verify sequential by client(callee).

I don't know why it must be sequential, why cannot just a session scope unique random ID?

**Comment by SunFulong** [comment]
@oberstet @estan @KSDaemon , would you explain this?

**Comment by oberstet** [comment]
I guess we should expand the topic "orderings" (and IDs) - again - in the spec ... provide more background ... all of this results from extensive discussions and experimentation ...

> I don't know why it must be sequential, why cannot just a session scope unique random ID?

here is one reason:

a) using session scoped sequential IDs defines a unique (total) ordering of all RPC related messages as sent by a single specific peer (the respective WAMP session), and
b) that ordering can be observed by all intermediate router nodes, as well as the other peer involved


> the fact that the total ordering is well defined and unique, and that it can be observed by intermediate router nodes, is of particular importance when using router-to-router messaging
